File: src/main/java/uk/gov/justice/digital/hmpps/oauth2server/security/UserPersonDetails.java
package uk.gov.justice.digital.hmpps.oauth2server.security;

import org.springframework.security.core.CredentialsContainer;
import org.springframework.security.core.userdetails.UserDetails;
import uk.gov.justice.digital.hmpps.oauth2server.auth.model.User;

public interface UserPersonDetails extends UserDetails, CredentialsContainer {
    String getUserId();

    String getName();

    String getFirstName();

    boolean isAdmin();

    String getAuthSource();

    User toUser();
}

File: src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/security/UserPersonDetails.kt
package uk.gov.justice.digital.hmpps.oauth2server.security

import org.springframework.security.core.CredentialsContainer
import org.springframework.security.core.userdetails.UserDetails
import uk.gov.justice.digital.hmpps.oauth2server.auth.model.User

interface UserPersonDetails : UserDetails, CredentialsContainer {
  val userId: String
  val name: String
  val firstName: String
  val isAdmin: Boolean
  val authSource: String
  fun toUser(): User
}
